homeassistant/components/pi_hole/sensor.py

```diff
@@ -89,6 +123,21 @@ def __init__(
     def native_value(self) -> StateType:
         """Return the state of the device."""
         try:
-            return round(self.api.data[self.entity_description.key], 2)  # type: ignore[no-any-return]
+            return round(get_nested(self.api.data, self.entity_description.key), 2)
         except TypeError:
-            return self.api.data[self.entity_description.key]  # type: ignore[no-any-return]
+            return get_nested(self.api.data, self.entity_description.key)
+
+
+def get_nested(data: Mapping[str, Any], key: str) -> float | int:
```

Comment:
Is there a more elegant solution for traversing the nested dict than dot notation above and this function?

Reply:
I wanted there to be one, but no. Someone who loves standards might say to use [JSON pointers](https://datatracker.ietf.org/doc/html/rfc6901), but there's really no point.